lib/rules/no-extra-parens.js

```diff
@@ -395,13 +413,14 @@ module.exports = {
          */
         function dryBinaryLogical(node) {
             const prec = precedence(node);
-            const shouldSkipLeft = NESTED_BINARY && (node.left.type === "BinaryExpression" || node.left.type === "LogicalExpression");
+            const shouldSkipLeft = (NESTED_BINARY && (node.left.type === "BinaryExpression" || node.left.type === "LogicalExpression")) ||
+              node.left.type === "UnaryExpression" && node.operator === "**";
```

Comment:
I mean I don't see how adding a precedence for `**` (which I will anyways, I'm not arguing it should be there) would change the code I added to deal with associativity.